templates/flightfact.cmds.erb

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+<% cmd_base="flight fact set #{@asset_data.name}" %>
+
+# General System Information
+<%= cmd_base %> 'Model' '<%= @asset_data.lshw.list.node.product %>'
+<%= cmd_base %> 'BIOS_Version' '<%= find_hashes_with_key_value(@asset_hash, 'id', 'firmware')&.first['version'] %>'
```

Comment:
This line doesn't look quite right to me.  If the firmware cannot be found it will raise an error. The following will prevent the error in that case:

```
...'<%= find_hashes_with_key_value(@asset_hash, 'id', 'firmware')&.first&.fetch('version') %>'
```

If the firmware cannot be found it will result in the following:

```
flight fact set NODE 'BIOS_Version' ''
```

I'm not sure if that's a good command to run or not.  We could set a default value if the firmware doesn't exist:

```
...'<%= find_hashes_with_key_value(@asset_hash, 'id', 'firmware')&.first&.fetch('version', 'UNKNOWN') %>'
```

If the firmware cannot be found it will result in the following:

```
flight fact set NODE 'BIOS_Version' 'UNKNOWN'
```

Reply:
Thanks for the feedback, the current method doesn't seem to fail at the moment, I've tested on cloud, metal and virtual nodes and all usually return a bios version. Will put this in place to handle edge-cases nonetheless.